test/unit/job_invocation_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_plugin_helper'
+
+describe JobInvocation do
+
+  let(:job_invocation) { FactoryGirl.build(:job_invocation) }
+  let(:template) { FactoryGirl.build(:job_template) }
+
+  context 'Able to be created' do
+    it { assert job_invocation.save }
+  end
+
+  context 'Requires targeting' do
+    before { job_invocation.targeting = nil }
+
+    it { refute job_invocation.save }
+  end
+
+  context 'has template invocations' do
+    before do
+      job_invocation.template_invocations << TemplateInvocation.new(:template => template)
+      job_invocation.save!
+    end
+
+    it { refute job_invocation.reload.template_invocations.empty? }
```

Comment:
this test seems like testing of ActiveRecord functionality, if you insist on this test, could you use FactoryGirl for TemplateInvocation building?